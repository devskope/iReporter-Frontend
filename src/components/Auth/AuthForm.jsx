/* eslint-disable react/no-array-index-key */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import AuthFormHeader from './AuthFormHeader';
import AuthFormLinks from './AuthFormLinks';
import AuthFormField from './AuthFormField';
import AuthMessageWidget from './AuthMessageWidget';

const AuthForm = ({
  authType,
  clearErrors,
  errors,
  fields,
  handleFormSubmit,
  handleInputChange,
}) => (
  <Fragment>
    <main className="auth">
      <form className="auth__form" onSubmit={handleFormSubmit}>
        <AuthFormHeader authType={authType} />
        <AuthMessageWidget errors={errors} clearErrors={clearErrors} />
        <div className="form-fields">
          {fields.map((field, i) => (
            <AuthFormField
              key={i}
              field={field}
              handleInputChange={handleInputChange}
            />
          ))}
          <button className="btn btn--submit" type="submit">
            {authType === 'LOGIN' ? 'Login' : 'Create account'}
          </button>
        </div>
        <AuthFormLinks authType={authType} />
      </form>
    </main>
  </Fragment>
);

AuthForm.propTypes = {
  authType: PropTypes.oneOf(['LOGIN', 'SIGNUP']).isRequired,
  errors: PropTypes.arrayOf(PropTypes.string).isRequired,
  fields: PropTypes.arrayOf(
    PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool]))
  ).isRequired,
  clearErrors: PropTypes.func.isRequired,
  handleFormSubmit: PropTypes.func.isRequired,
  handleInputChange: PropTypes.func.isRequired,
};

export default AuthForm;
