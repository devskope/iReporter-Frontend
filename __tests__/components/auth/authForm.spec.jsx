import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import AuthForm from '../../../src/components/Auth/AuthForm';

const setup = () => {
  const props = {
    authType: 'LOGIN',
    fields: [
      {
        label: 'Username',
        type: 'text',
        name: 'username',
        placeholder: 'Ahmed',
        required: true,
        value: 'abba',
      },
    ],
    handleFormSubmit: jest.fn(),
    handleInputChange: jest.fn(),
    clearErrors: jest.fn(),
    errors: ['error message', 'error message'],
  };
  return mount(
    <MemoryRouter>
      <AuthForm {...props} />
    </MemoryRouter>
  );
};

describe('AuthForm', () => {
  it('should render without crashing', () => {
    const wrapper = setup();

    expect(wrapper.find('.auth__message').exists()).toBe(true);
    expect(wrapper.find('.auth__message-list').exists()).toBe(true);
    expect(wrapper.find('.form-fields').exists()).toBe(true);
    expect(wrapper.find('label')).toHaveLength(1);
    expect(wrapper.find('.form-field').exists()).toBe(true);
    expect(wrapper.find('.btn--submit').exists()).toBe(true);
  });

  it('should match snapshot', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
