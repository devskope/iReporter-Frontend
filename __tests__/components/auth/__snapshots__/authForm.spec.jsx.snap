// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm should match snapshot 1`] = `ReactWrapper {}`;
